libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -2268,21 +2268,21 @@ WaveTrack::GetOneSampleView(sampleCount start, size_t length) const
       const auto clipStartTime = clip->GetPlayStartTime();
       if (t0 < clipStartTime) {
          const auto numSamples = TimeToLongSamples(clipStartTime - t0);
-         segments.push_back(AudioSegmentSampleView(numSamples));
+         segments.push_back(AudioSegmentSampleView{numSamples});
          t0 = clipStartTime;
+         length -= numSamples.as_size_t();
       }
       const auto clipT0 = t0 - clipStartTime;
       const auto clipS0 = TimeToLongSamples(clipT0);
-      // By substitution,
-      //    (len - length) / rate = t0 - clipT0,
-      // and since
-      //    t0 - clipT0 <= 0,
-      // then `length >= len`. `as_size_t` won't narrow.
-      const auto len = TimeToLongSamples(t1 - clipT0).as_size_t();
+      const auto len =
+          std::min((clip->GetPlaySamplesCount() - clipS0).as_size_t(), length);
```

Comment:
You might use `limitSampleBufferSize`

Reply:
In fact that would be more correct.  The narrowing to size_t here might sometimes make a small number.  Better to widen `length` before taking `min`